File: Assignment no. 36/Q3.c
#include<stdio.h>

typedef int BOOL;
#define TRUE 1
#define FALAE 0

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//  Name        :   StrNCmpX
//  Input       :   character address,character address, integer
//  Output      :   boolean
//  Description :   check weather first n characters of two strings are equal or not
//  Author      :   Nikhil
//  Date        :   25/10/2019
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

BOOL StrCmpX(char *src,char *dest,int iCnt)
{
    if((src==NULL)||(dest==NULL))
    {
        return;
    }
    while((*src!='\0')&&(*dest!='\0')&&(iCnt>=0))
    {
        if(*src!=*dest)
        {
            break;
        }
        src++;
        dest++;
        iCnt--;
    }
    if(iCnt==-1)
    {
        return TRUE;
    }
    else
    {
        return FALAE;
    }
}

int main()
{
    char Arr[50]="Marvelous Infosystems";
    char Brr[30]="Marvelous Infosystems";
    int iValue=0;

    printf("Enter first string:\n");
    scanf("%[^'\n']s",Arr);
    fflush(stdin);
    printf("Enter second string:\n");
    scanf("%[^'\n']s",Brr);
    printf("Enter number:\n");
    scanf("%d",&iValue);
    if(StrCmpX(Arr,Brr,iValue)==TRUE)
    {
        printf("Both strings are equal");
    }
    else
    {
        printf("Both strings are not equal");
    }

    return 0;
}

File: Assignment no. 60/Q3.java
import java.util.Scanner;

class Q3
{
	public static void main(String arg[])
	{
		Scanner sobj=new Scanner(System.in);
		System.out.println("Enter the first string :");
		String src=sobj.nextLine();
		System.out.println("Enter the second string :");
		String dest=sobj.nextLine();
		System.out.println("Enter the number of character to be compare:");
		int size=sobj.nextInt();
		StringDemo obj=new StringDemo();
		if(obj.strNCmpX(src,dest,size))
		{
			System.out.println("Strings are equal");
		}
		else
		{
			System.out.println("Strings are not equal");
		}
	}
}